### PR TITLE
#159340982 add user signup

### PR DESCRIPTION
**Pull request by damafeez**
#### What does this PR do?

- add user signup controller
- add P0ST /auth/signup route
- add passing POST /auth/signup test

#### Description of Task to be completed?
- password hashing
- more test cases

#### How should this be manually tested?
run `npm test`

#### Any background context you want to provide?
n/a

#### What are the relevant pivotal tracker stories?
#159340982

Questions:
None

### CODE REVIEW

File: server/controllers/user.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+const client = connection();
+client.connect();
+
+export async function signup(req, res) {
+  const requiredFields = ['fullName', 'username', 'password'];
+  try {
+    const { fullName, username, password } = req.body;
+    if (fullName && username && password) {
+      const authenticationQuery = `INSERT INTO authentication (username, password)
```
Comment (fidelisojeah) [inline server/controllers/user.js]: You should also insert into the users table as well

**Review by fidelisojeah — COMMENTED** [review]
Include the postgreSQL module in your travis: 
```yml
services:
  - postgresql
```
Increase your test coverage also

File: server/models/User.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    password,
+  }) {
+    this.fullName = fullName;
+    this.email = email;
+    this.username = username;
+    this.password = password;
+  }
+
+  async save() {
+    const authQuery = 'INSERT INTO authentication(username, password) VALUES($1, $2) RETURNING *';
```
Comment (fidelisojeah) [inline server/models/User.js:19]: Okay rather than just sending the password directly into the database, you should hash it
Good job though

**Review by fidelisojeah — CHANGES_REQUESTED** [review]

File: server/models/User.js (hunk 2 above)
Comment (damafeez) [reply server/models/User.js:19]: Yes I will. It's added in the `Description of Task to be completed?`. Thanks.

**Pull request merged**